examples/User_Type.jl

```diff
@@ -0,0 +1,43 @@
+using ODE
+using PyPlot
+
+typealias DataFloat Float64
+
+type Vec2
+  a::DataFloat
+  b::DataFloat
+end
+
+# Necessary for the solver
+Base.(:/)(x::Vec2, y::Real) = Vec2(x.a/y, x.b/y)
+Base.(:*)(y::Real, x::Vec2) = Vec2(y * x.a, y * x.b)
+Base.(:*)(x::Vec2, y::Real) = y*x
+Base.(:.*)(y::Real, x::Vec2) = y*x
+Base.(:+)(x::Vec2, y::Vec2) = Vec2(x.a + y.a, x.b + y.b)
+Base.(:-)(x::Vec2, y::Vec2) = Vec2(x.a - y.a, x.b - y.b)
+Base.norm(x::Vec2) = sqrt(x.a^2 + x.b^2)
+Base.zero(x::Type{Vec2}) = Vec2(zero(DataFloat), zero(DataFloat))
+Base.isnan(x::Vec2) = isnan(x.a) || isnan(x.b)
+
+function f(t, y)
+  (a, b) = (y.a, y.b)
+  Vec2(-b, a)
+end
+
+let
+  # Initial condtions
+  const start = Vec2(0.0, 0.1)
```

Comment:
These don't need to be `const` for performance as they are passed into the function as arguments.